tests/src/opaque_test_driver.c

```diff
@@ -0,0 +1,338 @@
+            mbedtls_mpi_write_binary( &ecp.d, key, bytes ) );
+
+        if( status == PSA_SUCCESS )
+            *key_length = bytes;
+
+        mbedtls_ecp_keypair_free( &ecp );
+        return( status );
+    }
+
+    else
+#endif /* MBEDTLS_ECP_C */
+        return( PSA_ERROR_NOT_SUPPORTED );
+
+    return( opaque_test_driver_import_key( attributes,
+                                           key_buffer,
+                                           key_buffer_length,
+                                           key,
+                                           key_size,
+                                           key_length ) );
+}
+
+psa_status_t opaque_test_driver_import_key(
+                                        const psa_key_attributes_t *attributes,
+                                        const uint8_t *in,
+                                        size_t in_length,
+                                        uint8_t *out,
+                                        size_t out_size,
+                                        size_t *out_length )
+{
+    OPQTD_VALIDATE_RET( attributes != NULL );
+    OPQTD_VALIDATE_RET( in != NULL );
+    OPQTD_VALIDATE_RET( out != NULL );
+    OPQTD_VALIDATE_RET( out_length != NULL );
+
+    if( ( psa_get_key_type( attributes ) != PSA_KEY_TYPE_AES ) &&
+        ( psa_get_key_type( attributes ) !=
+          PSA_KEY_TYPE_ECC_KEY_PAIR( PSA_ECC_CURVE_SECP_R1 ) ) )
+        return( PSA_ERROR_NOT_SUPPORTED );
+
+    if( ( psa_get_key_bits( attributes ) != 128 ) &&      // AES-128
+        ( psa_get_key_bits( attributes ) != 192 ) &&      // AES-192
+        ( psa_get_key_bits( attributes ) != 256 ) )       // AES-256
+        return( PSA_ERROR_NOT_SUPPORTED );
+
+    if( psa_get_key_bits( attributes ) != PSA_BYTES_TO_BITS( in_length ) )
+        return( PSA_ERROR_INVALID_ARGUMENT );
+
+    if( OPAQUE_TEST_DRIVER_KEYHEADER_SIZE + in_length > out_size )
+        return( PSA_ERROR_BUFFER_TOO_SMALL );
+
+    strcpy( (char *) out, OPAQUE_TEST_DRIVER_KEYHEADER );
+
+    /* Obscure key slightly. */
+    rot13( in, in_length, out + OPAQUE_TEST_DRIVER_KEYHEADER_SIZE );
+
+    *out_length = in_length + OPAQUE_TEST_DRIVER_KEYHEADER_SIZE;
+
+    return( PSA_SUCCESS );
+}
+
+psa_status_t opaque_test_driver_sign_hash(
+                                        const psa_key_attributes_t *attributes,
+                                        const uint8_t *key,
+                                        size_t key_length,
+                                        psa_algorithm_t alg,
+                                        const uint8_t *hash,
+                                        size_t hash_length,
+                                        uint8_t *signature,
+                                        size_t signature_size,
+                                        size_t *signature_length )
+{
+    #define OPQ_BUFSIZE 64
+    size_t key_buffer_length;
+    psa_key_handle_t handle = 0;
+    uint8_t key_buffer[OPQ_BUFSIZE];
```

Comment:
The right size would be `uint8_t key_buffer[PSA_EXPORT_KEY_PAIR_MAX_SIZE]` but unfortunately we haven't yet merged https://github.com/ARMmbed/mbedtls/pull/3386 that adds it to Mbed TLS.